lib/Backend/GlobOpt.cpp

```diff
@@ -9611,8 +9621,10 @@ GlobOpt::OptConstFoldUnary(
             }
             else
             {
-                isInt = false;
-                fValue = -(FloatConstType)INT32_MIN;
+                // disable const fold for Math.abs(INT32_MIN) because it causes dst to be float type which
+                // could be different with previous type spec result in LoopPrePass. The latter does int
+                // type spec for Math.abs(INT32_MIN).
+                return false;
```

Comment:
RejitReason::AggressiveIntTypeSpecDisabled might be too aggressive just because we cannot do typespec on constant INT32_MIN?

---
In reply to: [94697935](https://github.com/Microsoft/ChakraCore/pull/2267#discussion_r94697935) [](ancestors = 94697935)